src/styles/index.scss

```diff
@@ -1,3 +1,300 @@
+            bottom: 0;
+            left: 0;
+          }
+        }
+      }
+    }
+  }
+
+  &__logo-link,
+  &__logo-link img {
+    width: $logo-size;
+    height: auto;
+    display: block;
+  }
+}
+
+/* #endregion HEADER */
+
+/* #region CATALOG */
+.catalog {
+  display: grid;
+  width: 944px;
+
+  grid-template-columns: repeat(4, 200px); /* 4 колонки по 200px */
+  justify-content: center;
+  gap: 46px 48px;
+  margin: 50px auto;
+}
+
+@media (min-width: 1024px) {
+  .catalog {
+    width: 1024px;
+  }
+}
+
+@media (min-width: 1200px) {
+  .catalog {
+    width: 1200px;
+  }
+}
+
+/* #endregion CATALOG */
+
+/* #region CARD */
+
+@mixin flex-size-center($width, $height) {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: $width;
+  height: $height;
+}
+
+@mixin star-position {
+  width: 16px;
+  height: 16px;
+  margin-left: 4px;
+  background-position: center;
+  background-repeat: no-repeat;
+}
+
+// Міксин для стилю тексту
+@mixin text-style($font-size, $font-weight, $color) {
+  font-size: $font-size;
+  font-weight: $font-weight;
+  color: $color;
+}
+
+p {
+  margin: 0;
+}
+
+.card {
+  display: flex;
+  width: $card-width;
+  height: $card-height;
+  border: 1px solid #f3f3f3;
+  border-radius: 5px;
+
+  &:hover {
+    transform: scale(1.2);
+    transition: transform 300ms ease;
+  }
+
+  &__content {
+    width: $card-width;
+    height: $card-height;
+    border-radius: 5px;
+  }
+
+  &__image {
+    @include flex-size-center(160px, 134px);
+
+    background-image: url(../images/imac.jpeg);
+    margin-left: 19px;
+    margin-top: 32px;
+    background-size: cover; // або contain, в залежності від вашого бажання
+  }
+
+  &__title {
+    margin-top: 40px;
+    margin-left: $left-margin;
+    height: 36px;
+    width: 166px;
+
+    &-text {
+      @include text-style($text-size, 500, $text-color);
+
+      line-height: 18px;
+
+      &:hover {
+        color: #34568b;
+        transition: color 300ms ease;
+      }
+    }
+  }
+
+  &__code {
+    width: 98px;
+    height: 14px;
+    margin-top: 4px;
+    margin-left: $left-margin;
+    line-height: 14px;
+
+    &-text {
+      font-size: 10px;
+      font-weight: 400;
+      color: $subtext-color;
+    }
+  }
+
+  &__rating {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    margin-left: 16px;
+    margin-top: 16px;
+    flex-wrap: wrap;
+    width: 166px;
+
+    &-stars {
+      display: flex;
+    }
+
+    // Цикл для стилізації зірок (діти класу &__rating)
+    &-star {
+      background-image: url(../images/star.svg); // Зображення за замовчуванням для всіх зірок
+
+      // Цикл для стилізації активних зірок
+      @for $i from 1 through $stars-count {
+        &:nth-child(#{$i}) {
+          @if $i <= $active-stars {
+            background-image: url(../images/star-active.svg); // Заповнена зірка для активних
+          }
+
+          @include star-position; // Позиціонування зірки
+        }
+      }
```

Comment:
It's good that you're using a loop to generate the stars. However, it's not very efficient to set the background image in each iteration. Consider setting the default star image outside of the loop, and only change it for the active stars inside the loop.